File: SampleCode/RestApi/Clients/Net.RestSharp/EmbyClient.Dotnet/Model/RemoteImageInfo.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// Class RemoteImageInfo  
    /// </summary>
    [DataContract]
        public partial class RemoteImageInfo :  IEquatable<RemoteImageInfo>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="RemoteImageInfo" /> class.
        /// </summary>
        /// <param name="providerName">The name of the provider..</param>
        /// <param name="url">The URL..</param>
        /// <param name="thumbnailUrl">A url used for previewing a smaller version.</param>
        /// <param name="height">The height..</param>
        /// <param name="width">The width..</param>
        /// <param name="communityRating">The community rating..</param>
        /// <param name="voteCount">The vote count..</param>
        /// <param name="language">The language..</param>
        /// <param name="displayLanguage">displayLanguage.</param>
        /// <param name="type">type.</param>
        /// <param name="ratingType">ratingType.</param>
        public RemoteImageInfo(string providerName = default(string), string url = default(string), string thumbnailUrl = default(string), int? height = default(int?), int? width = default(int?), double? communityRating = default(double?), int? voteCount = default(int?), string language = default(string), string displayLanguage = default(string), ImageType type = default(ImageType), RatingType ratingType = default(RatingType))
        {
            this.ProviderName = providerName;
            this.Url = url;
            this.ThumbnailUrl = thumbnailUrl;
            this.Height = height;
            this.Width = width;
            this.CommunityRating = communityRating;
            this.VoteCount = voteCount;
            this.Language = language;
            this.DisplayLanguage = displayLanguage;
            this.Type = type;
            this.RatingType = ratingType;
        }
        
        /// <summary>
        /// The name of the provider.
        /// </summary>
        /// <value>The name of the provider.</value>
        [DataMember(Name="ProviderName", EmitDefaultValue=false)]
        public string ProviderName { get; set; }

        /// <summary>
        /// The URL.
        /// </summary>
        /// <value>The URL.</value>
        [DataMember(Name="Url", EmitDefaultValue=false)]
        public string Url { get; set; }

        /// <summary>
        /// A url used for previewing a smaller version
        /// </summary>
        /// <value>A url used for previewing a smaller version</value>
        [DataMember(Name="ThumbnailUrl", EmitDefaultValue=false)]
        public string ThumbnailUrl { get; set; }

        /// <summary>
        /// The height.
        /// </summary>
        /// <value>The height.</value>
        [DataMember(Name="Height", EmitDefaultValue=false)]
        public int? Height { get; set; }

        /// <summary>
        /// The width.
        /// </summary>
        /// <value>The width.</value>
        [DataMember(Name="Width", EmitDefaultValue=false)]
        public int? Width { get; set; }

        /// <summary>
        /// The community rating.
        /// </summary>
        /// <value>The community rating.</value>
        [DataMember(Name="CommunityRating", EmitDefaultValue=false)]
        public double? CommunityRating { get; set; }

        /// <summary>
        /// The vote count.
        /// </summary>
        /// <value>The vote count.</value>
        [DataMember(Name="VoteCount", EmitDefaultValue=false)]
        public int? VoteCount { get; set; }

        /// <summary>
        /// The language.
        /// </summary>
        /// <value>The language.</value>
        [DataMember(Name="Language", EmitDefaultValue=false)]
        public string Language { get; set; }

        /// <summary>
        /// Gets or Sets DisplayLanguage
        /// </summary>
        [DataMember(Name="DisplayLanguage", EmitDefaultValue=false)]
        public string DisplayLanguage { get; set; }

        /// <summary>
        /// Gets or Sets Type
        /// </summary>
        [DataMember(Name="Type", EmitDefaultValue=false)]
        public ImageType Type { get; set; }

        /// <summary>
        /// Gets or Sets RatingType
        /// </summary>
        [DataMember(Name="RatingType", EmitDefaultValue=false)]
        public RatingType RatingType { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class RemoteImageInfo {\n");
            sb.Append("  ProviderName: ").Append(ProviderName).Append("\n");
            sb.Append("  Url: ").Append(Url).Append("\n");
            sb.Append("  ThumbnailUrl: ").Append(ThumbnailUrl).Append("\n");
            sb.Append("  Height: ").Append(Height).Append("\n");
            sb.Append("  Width: ").Append(Width).Append("\n");
            sb.Append("  CommunityRating: ").Append(CommunityRating).Append("\n");
            sb.Append("  VoteCount: ").Append(VoteCount).Append("\n");
            sb.Append("  Language: ").Append(Language).Append("\n");
            sb.Append("  DisplayLanguage: ").Append(DisplayLanguage).Append("\n");
            sb.Append("  Type: ").Append(Type).Append("\n");
            sb.Append("  RatingType: ").Append(RatingType).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as RemoteImageInfo);
        }

        /// <summary>
        /// Returns true if RemoteImageInfo instances are equal
        /// </summary>
        /// <param name="input">Instance of RemoteImageInfo to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(RemoteImageInfo input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.ProviderName == input.ProviderName ||
                    (this.ProviderName != null &&
                    this.ProviderName.Equals(input.ProviderName))
                ) && 
                (
                    this.Url == input.Url ||
                    (this.Url != null &&
                    this.Url.Equals(input.Url))
                ) && 
                (
                    this.ThumbnailUrl == input.ThumbnailUrl ||
                    (this.ThumbnailUrl != null &&
                    this.ThumbnailUrl.Equals(input.ThumbnailUrl))
                ) && 
                (
                    this.Height == input.Height ||
                    (this.Height != null &&
                    this.Height.Equals(input.Height))
                ) && 
                (
                    this.Width == input.Width ||
                    (this.Width != null &&
                    this.Width.Equals(input.Width))
                ) && 
                (
                    this.CommunityRating == input.CommunityRating ||
                    (this.CommunityRating != null &&
                    this.CommunityRating.Equals(input.CommunityRating))
                ) && 
                (
                    this.VoteCount == input.VoteCount ||
                    (this.VoteCount != null &&
                    this.VoteCount.Equals(input.VoteCount))
                ) && 
                (
                    this.Language == input.Language ||
                    (this.Language != null &&
                    this.Language.Equals(input.Language))
                ) && 
                (
                    this.DisplayLanguage == input.DisplayLanguage ||
                    (this.DisplayLanguage != null &&
                    this.DisplayLanguage.Equals(input.DisplayLanguage))
                ) && 
                (
                    this.Type == input.Type ||
                    (this.Type != null &&
                    this.Type.Equals(input.Type))
                ) && 
                (
                    this.RatingType == input.RatingType ||
                    (this.RatingType != null &&
                    this.RatingType.Equals(input.RatingType))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.ProviderName != null)
                    hashCode = hashCode * 59 + this.ProviderName.GetHashCode();
                if (this.Url != null)
                    hashCode = hashCode * 59 + this.Url.GetHashCode();
                if (this.ThumbnailUrl != null)
                    hashCode = hashCode * 59 + this.ThumbnailUrl.GetHashCode();
                if (this.Height != null)
                    hashCode = hashCode * 59 + this.Height.GetHashCode();
                if (this.Width != null)
                    hashCode = hashCode * 59 + this.Width.GetHashCode();
                if (this.CommunityRating != null)
                    hashCode = hashCode * 59 + this.CommunityRating.GetHashCode();
                if (this.VoteCount != null)
                    hashCode = hashCode * 59 + this.VoteCount.GetHashCode();
                if (this.Language != null)
                    hashCode = hashCode * 59 + this.Language.GetHashCode();
                if (this.DisplayLanguage != null)
                    hashCode = hashCode * 59 + this.DisplayLanguage.GetHashCode();
                if (this.Type != null)
                    hashCode = hashCode * 59 + this.Type.GetHashCode();
                if (this.RatingType != null)
                    hashCode = hashCode * 59 + this.RatingType.GetHashCode();
                return hashCode;
            }
        }

    }
}

File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/RemoteImageInfo.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.client.model.ImageType;
import io.swagger.client.model.RatingType;
import io.swagger.v3.oas.annotations.media.Schema;
import java.io.IOException;
/**
 * Class RemoteImageInfo  
 */
@Schema(description = "Class RemoteImageInfo  ")

public class RemoteImageInfo {
  @SerializedName("ProviderName")
  private String providerName = null;

  @SerializedName("Url")
  private String url = null;

  @SerializedName("ThumbnailUrl")
  private String thumbnailUrl = null;

  @SerializedName("Height")
  private Integer height = null;

  @SerializedName("Width")
  private Integer width = null;

  @SerializedName("CommunityRating")
  private Double communityRating = null;

  @SerializedName("VoteCount")
  private Integer voteCount = null;

  @SerializedName("Language")
  private String language = null;

  @SerializedName("DisplayLanguage")
  private String displayLanguage = null;

  @SerializedName("Type")
  private ImageType type = null;

  @SerializedName("RatingType")
  private RatingType ratingType = null;

  public RemoteImageInfo providerName(String providerName) {
    this.providerName = providerName;
    return this;
  }

   /**
   * The name of the provider.
   * @return providerName
  **/
  @Schema(description = "The name of the provider.")
  public String getProviderName() {
    return providerName;
  }

  public void setProviderName(String providerName) {
    this.providerName = providerName;
  }

  public RemoteImageInfo url(String url) {
    this.url = url;
    return this;
  }

   /**
   * The URL.
   * @return url
  **/
  @Schema(description = "The URL.")
  public String getUrl() {
    return url;
  }

  public void setUrl(String url) {
    this.url = url;
  }

  public RemoteImageInfo thumbnailUrl(String thumbnailUrl) {
    this.thumbnailUrl = thumbnailUrl;
    return this;
  }

   /**
   * A url used for previewing a smaller version
   * @return thumbnailUrl
  **/
  @Schema(description = "A url used for previewing a smaller version")
  public String getThumbnailUrl() {
    return thumbnailUrl;
  }

  public void setThumbnailUrl(String thumbnailUrl) {
    this.thumbnailUrl = thumbnailUrl;
  }

  public RemoteImageInfo height(Integer height) {
    this.height = height;
    return this;
  }

   /**
   * The height.
   * @return height
  **/
  @Schema(description = "The height.")
  public Integer getHeight() {
    return height;
  }

  public void setHeight(Integer height) {
    this.height = height;
  }

  public RemoteImageInfo width(Integer width) {
    this.width = width;
    return this;
  }

   /**
   * The width.
   * @return width
  **/
  @Schema(description = "The width.")
  public Integer getWidth() {
    return width;
  }

  public void setWidth(Integer width) {
    this.width = width;
  }

  public RemoteImageInfo communityRating(Double communityRating) {
    this.communityRating = communityRating;
    return this;
  }

   /**
   * The community rating.
   * @return communityRating
  **/
  @Schema(description = "The community rating.")
  public Double getCommunityRating() {
    return communityRating;
  }

  public void setCommunityRating(Double communityRating) {
    this.communityRating = communityRating;
  }

  public RemoteImageInfo voteCount(Integer voteCount) {
    this.voteCount = voteCount;
    return this;
  }

   /**
   * The vote count.
   * @return voteCount
  **/
  @Schema(description = "The vote count.")
  public Integer getVoteCount() {
    return voteCount;
  }

  public void setVoteCount(Integer voteCount) {
    this.voteCount = voteCount;
  }

  public RemoteImageInfo language(String language) {
    this.language = language;
    return this;
  }

   /**
   * The language.
   * @return language
  **/
  @Schema(description = "The language.")
  public String getLanguage() {
    return language;
  }

  public void setLanguage(String language) {
    this.language = language;
  }

  public RemoteImageInfo displayLanguage(String displayLanguage) {
    this.displayLanguage = displayLanguage;
    return this;
  }

   /**
   * Get displayLanguage
   * @return displayLanguage
  **/
  @Schema(description = "")
  public String getDisplayLanguage() {
    return displayLanguage;
  }

  public void setDisplayLanguage(String displayLanguage) {
    this.displayLanguage = displayLanguage;
  }

  public RemoteImageInfo type(ImageType type) {
    this.type = type;
    return this;
  }

   /**
   * Get type
   * @return type
  **/
  @Schema(description = "")
  public ImageType getType() {
    return type;
  }

  public void setType(ImageType type) {
    this.type = type;
  }

  public RemoteImageInfo ratingType(RatingType ratingType) {
    this.ratingType = ratingType;
    return this;
  }

   /**
   * Get ratingType
   * @return ratingType
  **/
  @Schema(description = "")
  public RatingType getRatingType() {
    return ratingType;
  }

  public void setRatingType(RatingType ratingType) {
    this.ratingType = ratingType;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    RemoteImageInfo remoteImageInfo = (RemoteImageInfo) o;
    return Objects.equals(this.providerName, remoteImageInfo.providerName) &&
        Objects.equals(this.url, remoteImageInfo.url) &&
        Objects.equals(this.thumbnailUrl, remoteImageInfo.thumbnailUrl) &&
        Objects.equals(this.height, remoteImageInfo.height) &&
        Objects.equals(this.width, remoteImageInfo.width) &&
        Objects.equals(this.communityRating, remoteImageInfo.communityRating) &&
        Objects.equals(this.voteCount, remoteImageInfo.voteCount) &&
        Objects.equals(this.language, remoteImageInfo.language) &&
        Objects.equals(this.displayLanguage, remoteImageInfo.displayLanguage) &&
        Objects.equals(this.type, remoteImageInfo.type) &&
        Objects.equals(this.ratingType, remoteImageInfo.ratingType);
  }

  @Override
  public int hashCode() {
    return Objects.hash(providerName, url, thumbnailUrl, height, width, communityRating, voteCount, language, displayLanguage, type, ratingType);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class RemoteImageInfo {\n");
    
    sb.append("    providerName: ").append(toIndentedString(providerName)).append("\n");
    sb.append("    url: ").append(toIndentedString(url)).append("\n");
    sb.append("    thumbnailUrl: ").append(toIndentedString(thumbnailUrl)).append("\n");
    sb.append("    height: ").append(toIndentedString(height)).append("\n");
    sb.append("    width: ").append(toIndentedString(width)).append("\n");
    sb.append("    communityRating: ").append(toIndentedString(communityRating)).append("\n");
    sb.append("    voteCount: ").append(toIndentedString(voteCount)).append("\n");
    sb.append("    language: ").append(toIndentedString(language)).append("\n");
    sb.append("    displayLanguage: ").append(toIndentedString(displayLanguage)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    ratingType: ").append(toIndentedString(ratingType)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
